testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -105,6 +113,78 @@ def test_check_set_unit_nolength(self, D):
         with pytest.raises(ValueError):
             D._check_set_unit(units)
 
+    def test_check_set_unit_density_none(self, D):
+        units = {'density': None}
+        assert D._check_set_unit(units) == None
+
+    def test_check_set_unit_density_none(self, D):
+        units = {'density': None}
```

Comment:
The test logic here isn't clear to me.

1) You start with units of {'length': 'A', 'density': 'A^{-3}'}
2) Then you set units to None again
3) Then you delete the density units
4) Then you test that passing an empty dictionary will yield units of None for 'density'

The only thing that properly gets tested here is number 4 in that list (3 is kinda meant to test 4). What's the point for actions 1 and 2, why not just delete the units from the start? If the point is to check what happens when you pass None, then should there be a second assert?

Reply:
Thanks, _I think_ I followed your suggestions and change the test:
1. deleted units, 
2. assert units,
3. assert check_set_units.